core/src/main/kotlin/knote/NotebookManagerImpl.kt

```diff
@@ -55,32 +55,39 @@ internal object NotebookManagerImpl : NotebookManager, KLogging() {
             return notebook
         }
 
-        notebook.pageManager = PageManagerImpl(notebook, host)
+        if(notebook.pageManager == null) {
+            notebook.pageManager = PageManagerImpl(notebook, host)
+        }
         return notebook
     }
 
-    override fun getPageManager(): PageManager? {
+    override val pageManager: PageManager
+    get() {
         val notebookId = KNote.notebookId
         val notebook = compileNotebookCached() as? NotebookImpl ?: run {
             logger.error("cannot load notebook $notebookId")
-            return null
+            throw IllegalStateException("cannot load notebook for $notebookId")
         }
-        return notebook.pageManager
-            ?: PageManagerImpl(notebook, host).also {
+        val pageManager = notebook.pageManager.let {
+            it ?: PageManagerImpl(notebook, host).also {
```

Comment:
in what case would this happen?

Reply:
in case the notebook failed compilation the pagemanager would also fail to set properly, the problem is that the pageManager needs the compiled noteboook to get initialized otherwise it crashes, so it needs to wait to get initialized